app/services/permissions_checker.rb

```diff
@@ -61,7 +62,8 @@ def authorize_manage_rooms
   def authorize_shared_room
     return false if @friendly_id.blank? && @record_id.blank?
 
-    @current_user.shared_rooms.exists?(friendly_id: @friendly_id) ||
+    SharedAccess.joins(:shared_room).find_by(user_id: @current_user.id, 'shared_room.friendly_id': @friendly_id).present? ||
```

Comment:
Why the change here?

Reply:
Doesn't this solve the "delete another shared access" issue actually?
It checks if the user is trying to do an action on the SharedAccess specifically, instead of just checking if the room is a shared room of the user.